lib/src/logic.dart

```diff
@@ -562,6 +562,9 @@ class Logic {
   /// Logical right-shift.
   Logic operator >>>(dynamic other) => RShift(this, other).out;
 
+  /// Unary clog2
+  Logic clog2() => Clog2Unary(this).out;
```

Comment:
Upon doing some googling, it is unclear to me whether `$clog2` is a synthesizable function in SystemVerilog.  It seems legal for constant value inputs and for width calculations, but I don't see any dynamic signals fed into `$clog2` in a synthesizable design.

If it turns out to _not_ be synthesizable, then at the very least we need to document here that it should not be used for cases where it is desired to be synthesized, or else we need to generate synthesizable hardware that actually calculates the right value.  If we leave it as a non-synthesizable option, do we want to keep it in ROHD?  It could be useful for testbench-related module generation perhaps.

Thoughts?

Reply:
Upon googling, I found out that clog2 is ‘synthesizable with limitations’.
Input to `$clog2` must be static-elaboration time constant and cannot be signal. Hence it cannot be used to model actual hardware. 
Reference: [https://www.csl.cornell.edu/courses/ece5745/handouts/ece5745-verilog-rules.pdf](url)

Considering these, maybe it's not required in ROHD. As a new user, it may be confusing to have clog2 option in ROHD which is actually non-synthesizable for signal. But if it is really required, we can keep by explicitly mentioning that it is synthesizable for constant value inputs only.

Let me know if we need to proceed with this PR.

Reply:
Thanks for the link, that seems like a good reference.

Since there's not a legal way to pass a signal to it, perhaps it's not appropriate to have a `Module` that accepts `Logic`s to compute it.  However, the functionality still seems valuable for `LogicValue` to have.  So I think it's worth proceeding with the PR, but unfortunately I think we should omit the part that applies to synthesizable hardware generation (`Module` and `Logic` stuff) and just keep the `LogicValue` addition.  How does that sound to you?

Reply:
Yes, it would better to proceed with the PR without synthesizable `Module` that accepts `Logic`s.